backend/tests/conftest.py

```diff
@@ -164,6 +222,56 @@ async def title(title_dict):
 async def title_with_language(title, language_eng):
     await title.languages.add(language_eng)
     yield title
+    await title.languages.clear()
+    await title.tags.clear()
+    await title.metadata.clear()
+    await title.delete()
+
+
+@pytest.fixture(scope="function")
+@pytest.mark.asyncio
+async def title_tag():
+    tag = await TitleTag.objects.create(name="wikipedia")
+    yield tag
+    await tag.delete()
+
+
+@pytest.fixture(scope="function")
+@pytest.mark.asyncio
+async def title_with_data(title, language_eng, title_tag, book_dict):
+    await title.languages.add(language_eng)
+    # await title.tags.add(title_tag)
```

Comment:
that should be restored

Reply:
this makes me realize that it is better to have static value in the test, rather than dynamic, because otherwise the tests don't fail because they are checking and asserting from the same data

so I am going to assert like this
```python
    assert response.json() == {
        "ident": title_with_data.ident,
        "languages": ["eng"],
        "tags": ["wikipedia"],
        "metadata": book_dict["metadata"],
    }
```

this way if we don't add tag to the title our test will fail

Reply:
this makes me realize that it is better to have static value in the test, rather than dynamic, because otherwise the tests don't fail because they are checking and asserting from the same data

so I am going to assert like this
```python
    assert response.json() == {
        "ident": title_with_data.ident,
        "languages": ["eng"],
        "tags": ["wikipedia"],
        "metadata": book_dict["metadata"],
    }
```

this way if we don't add tag to the title our test will fail

What do you say? @rgaudin

Reply:
Ah sorry I missed this comment.

- we don't want to maintain static values at different places. this will be hell to maintain.
- the model-instance fixtures are to provide fake data to the DB so we can query it easily
- we should'nt use those model-instance fixtures as source to assert ; as you mentioned, it's flawed.
- the model-instance fixtures are/should-be built from static values though (like the book_dict). We should use those to compare endpoint-returned data with what we got.

Reply:
@rgaudin just saw this, I agree, fixing this now